domain/application/state/types.go

```diff
@@ -637,3 +637,20 @@ type applicationCharmDownloadInfo struct {
 	DownloadSize       int64  `db:"download_size"`
 	SourceID           int    `db:"source_id"`
 }
+
+// resourceToAdd represents a resource we want to add into the state.
+// It is used to insert data in resource and application_resource tables.
```

Comment:
This table is also overloaded for adding a resource. The resource does not contain application_uuid, state_name, nor origin_type_name. An application_resource only contains an application_uuid an a resource_uuid.

Reply:
I don't get what is your concern. "It is used to insert data in **resource and application_resource** tables" => maybe you read "It is used to insert *resource* in **application_resource** tables" ?